clouddriver-core/src/main/java/com/netflix/spinnaker/clouddriver/model/NamespaceProvider.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright 2021 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package com.netflix.spinnaker.clouddriver.model;
+
+import com.netflix.spinnaker.clouddriver.documentation.Empty;
+import java.util.Collection;
+
+public interface NamespaceProvider {
```

Comment:
We're using the term `region` in many places (coming from AWS), `location` in others. I'm hesitant to add one more term. I think it only matches Kubernetes (?)

Reply:
Yes, Namespace in its literal sense is a k8 object but the abstraction is more or less what I was trying to accomplish. Namespace being a logical grouping around some resource. This happens to be a 1-1 with k8 and very close to 1-1 with CF (Org + Space combo). These are also dynamic in nature and can be created/updated/deleted. The tight coupling between the credentials and these "namespaces" is the main reason for this PR. Regions being more static might not make sense in this light

I'm open to calling it something different if you have a better abstraction in mind?

Reply:
I was thinking `location` to be consistent. It's not a very strongly held feeling on my side, I just wanted you to be aware that it's yet another naming scheme.  Alternative idea: we could define: `/credentials/kubernetes/[account name]/namespaces` for Kubernetes, `/credentials/cloudfoundry/[account name]/spaces` for CF, etc. I guess it depends how you plan to use this in Deck.